res/controllers/Traktor-Kontrol-S4-MK2-hid-scripts.js

```diff
@@ -0,0 +1,1275 @@
+/****************************************************************/
+/*      Traktor Kontrol S4 MK2 HID controller script v1.00      */
+/*      Copyright (C) 2015, the Mixxx Team                      */
+/*      but feel free to tweak this to your heart's content!    */
+/*      For Mixxx version 2.0                                   */
+/****************************************************************/
+
+
+// TODO:
+// * Loop size readout / selection
+// * Find a use for snap / master / other unused buttons?
+
+TraktorS4MK2 = new function() {
```

Comment:
the only reason I had created a "construct" method was to get the user-visible configuration closer to the top.  With the new solution I was able to revert to the previous way.